opensearch/src/test/java/org/opensearch/sql/opensearch/data/type/OpenSearchDateTypeTest.java

```diff
@@ -259,4 +263,158 @@ public void check_if_date_type_compatible() {
+    OpenSearchDateType dateType = OpenSearchDateType.of(format);
+    ZonedDateTime zonedDateTime = dateType.getParsedDateTime(timestamp);
+
+    assertEquals("2019-03-23T21:34:46", dateType.getFormattedDate(zonedDateTime.toInstant()));
+    assertEquals(LocalDate.parse("2019-03-23"), zonedDateTime.toLocalDate());
+    assertEquals(LocalTime.parse("21:34:46"), zonedDateTime.toLocalTime());
+    assertFalse(dateType.hasNoFormatter());
+  }
+
+  @Test
+  void test_openSearch_datetime_with_default_formatter() {
+    String timestamp = "2019-03-23T21:34:46";
+    OpenSearchDateType dateType = OpenSearchDateType.of(TIMESTAMP);
+    ZonedDateTime zonedDateTime = dateType.getParsedDateTime(timestamp);
+    // formatted using OpenSearch default formatter
+    assertEquals("2019-03-23T21:34:46Z", dateType.getFormattedDate(zonedDateTime.toInstant()));
+    assertEquals(LocalDate.parse("2019-03-23"), zonedDateTime.toLocalDate());
+    assertEquals(LocalTime.parse("21:34:46"), zonedDateTime.toLocalTime());
+    assertTrue(dateType.hasNoFormatter());
+  }
+
+  @Test
+  void test_invalid_date_with_named_formatter() {
+    // Incorrect date
+    String timestamp = "2019-23-23";
+    String format = "strict_date_hour_minute_second";
+    OpenSearchDateType dateType = OpenSearchDateType.of(format);
+    ZonedDateTime zonedDateTime = dateType.getParsedDateTime(timestamp);
+    assertNull(zonedDateTime);
+    assertFalse(dateType.hasNoFormatter());
+  }
+
+  @Test
+  void test_invalid_time_with_custom_formatter() {
+    String timestamp = "invalid-timestamp";
+    List<String> formats = Arrays.asList("yyyy/MM/dd'T'HH:mm:ssX", "yyyy-MM-dd'T'HH:mm:ssX");
+    OpenSearchDateType dateType = OpenSearchDateType.of(String.join(" || ", formats));
+    ZonedDateTime zonedDateTime = dateType.getParsedDateTime(timestamp);
+    assertNull(zonedDateTime);
+    assertFalse(dateType.hasNoFormatter());
+  }
+
+  @Test
+  void test_epoch_datetime_formatter() {
+    long epochTimestamp = 1636390800000L; // Corresponds to "2021-11-08T17:00:00Z"
+    String format = "epoch_millis";
+    OpenSearchDateType dateType = OpenSearchDateType.of(format);
+    ZonedDateTime zonedDateTime = dateType.getParsedDateTime(String.valueOf(epochTimestamp));
+
+    assertEquals(Long.toString(epochTimestamp), dateType.getFormattedDate(zonedDateTime));
+    assertEquals(LocalDate.parse("2021-11-08"), zonedDateTime.toLocalDate());
+    assertEquals(LocalTime.parse("17:00:00"), zonedDateTime.toLocalTime());
+    assertFalse(dateType.hasNoFormatter());
+  }
+
+  @Test
+  void test_timeStamp_format_with_default_formatters() {
+    String timestamp = "2021-11-08 17:00:00";
+    String format = "strict_date_time_no_millis || epoch_millis";
+    OpenSearchDateType dateType = OpenSearchDateType.of(format);
+    assertNull(dateType.getParsedDateTime(timestamp));
+  }
+
+  @Test
+  void test_valid_date_with_custom_formatter() {
+    String dateString = "2021-11-08";
+    String format = "yyyy-MM-dd";
```

Comment:
Could you test some date strings such as `23-Oct-17`?

Reply:
Done